assets/global.js

```diff
@@ -1273,3 +1273,161 @@ class ProductRecommendations extends HTMLElement {
 }
 
 customElements.define('product-recommendations', ProductRecommendations);
+
+class discountCart extends HTMLElement {
+  constructor() {
+    super();
+
+    const serializedArray = localStorage.getItem('discount_codes');
+    this.discountCodesArray = serializedArray ? JSON.parse(serializedArray) : [];
+    this.applyDiscountButton = this.querySelector('#apply_discount');
+    this.discountCodeField = this.querySelector('#discount_code');
+    this.discountCodeArray = this.discountCodesArray ? this.discountCodesArray : [];
+    this.discountWrapper = this.querySelector(".discount__wrapper");
+
+    if (this.discountCodesArray.length > 0 ) {
+      this.applyDiscount(true);
+    }
+
+    this.applyDiscountButton.addEventListener('click', () => this.applyDiscount());
+    this.removeDiscount = this.removeDiscount.bind(this);
+  }
+
+  applyDiscount (remItem) {
+    let discountCode = this.discountCodeField.value.trim();
+    this.discountCodeField.value = "";
+    this.discountWrapper.classList.remove("discount-error");
+    let discountCodeUnique = [];
+
+    if (discountCode || remItem ) {
+      if (remItem) {
+        discountCodeUnique = new Set(this.discountCodeArray);
+        discountCodeUnique = Array.from(discountCodeUnique)
+        discountCodeUnique = discountCodeUnique.filter(function(e) { return e !== remItem })
+        this.discountCodeArray = discountCodeUnique;
+
+        if (this.discountCodeArray.length === 0) {
+          localStorage.setItem('discount_codes', "");
+        }
+      } else {
+        this.discountCodeArray.push(discountCode);
+
+        if (this.discountCodeArray.length > 0 ) {
+          discountCodeUnique = new Set(this.discountCodeArray);
+          discountCodeUnique = Array.from(discountCodeUnique);
+          this.discountCodeArray = discountCodeUnique;
+        }
+      }
+
+      const cartToken = `gid://shopify/Cart/${document.cookie.split('; ').find(row => row.startsWith('cart=')).split('=')[1]}`;
```

Comment:
Write the cookie into a variable and then use the variable here. Also you can find getCookie function and then you don't need to split the string